packages/builders/src/interactions/commands/chatInput/mixins/ApplicationCommandOptionChannelTypesMixin.ts

```diff
@@ -0,0 +1,52 @@
+import { ChannelType, type APIApplicationCommandChannelOption } from 'discord-api-types/v10';
+import { normalizeArray, type RestOrArray } from '../../../../util/normalizeArray';
+
+export const applicationCommandOptionAllowedChannelTypes = [
+	ChannelType.GuildText,
+	ChannelType.GuildVoice,
+	ChannelType.GuildCategory,
+	ChannelType.GuildAnnouncement,
+	ChannelType.AnnouncementThread,
+	ChannelType.PublicThread,
+	ChannelType.PrivateThread,
+	ChannelType.GuildStageVoice,
+	ChannelType.GuildForum,
+	ChannelType.GuildMedia,
+] as const;
+
+/**
+ * Allowed channel types used for a channel option.
+ */
+export type ApplicationCommandOptionAllowedChannelTypes = (typeof applicationCommandOptionAllowedChannelTypes)[number];
```

Comment:
That doesn't seem right. It breaks the moment you add a new channel type. Best we can do here is type out the stuff in the array as a union if it's important for docs, otherwise, I'd keep it as-is.